presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -265,14 +269,31 @@ public PrestoSparkQueryExecutionFactory(
     private void checkPermissions(QueryId queryId, SessionContext sessionContext)
     {
         Identity identity = sessionContext.getIdentity();
-        accessControl.checkCanSetUser(
+        // When selectAuthorizedIdentity API is not enabled, only check delegation permission
+        if (!securityConfig.isAuthorizedIdentitySelectionEnabled()) {
+            accessControl.checkCanSetUser(
+                    identity,
+                    new AccessControlContext(
+                            queryId,
+                            Optional.ofNullable(sessionContext.getClientInfo()),
+                            Optional.ofNullable(sessionContext.getSource())),
+                    identity.getPrincipal(),
+                    identity.getUser());
+            return;
+        }
+
+        // When selectAuthorizedIdentity API is enabled,
+        // 1. Check the delegation permission, which is inside the API call
+        // 2. Select the authorized user
+        // 3. TODO: UPDATE the identity of the session according to the authorized identity from the API call
+        accessControl.selectAuthorizedIdentity(
```

Comment:
I am wondering why we are not updating the identity in this change? Are we doing this to collect data in first phase?

Even if that is the case, and we want to do it in two phase, we should log the returned identity in logs or somewhere.

Reply:
This is aligned with the rest of the Presto codebase, https://github.com/prestodb/presto/blob/c5b3bedc86e05fca9fbe7edd578abb88fae4246b/presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java#L350

Reply:
I see what you mean. I noticed a similar change has been made for Presto in #18439. Let me replicate that here.

Reply:
In general don't try to replicate PR, try to replicate logic from latest codebase as there may have been multiple commits/PR for a functionality.